spec/lib/pre_assembly/batch_spec.rb

```diff
@@ -59,52 +59,73 @@
 
       let(:yaml) { YAML.load_file(batch.progress_log_file) }
 
-      it 'logs an error' do
-        batch.process_digital_objects
+      it 'logs an error and returns false' do
+        expect(batch.process_digital_objects).to be false
         expect(yaml[:status]).to eq 'error'
         expect(yaml[:message]).to eq "can't be opened for a new version; cannot re-accession when version > 1 unless object can be opened"
       end
     end
 
     context 'when there are objects that do not complete pre_assemble' do
       before do
-        allow(batch.digital_objects[0]).to receive(:pre_assemble)
-        allow(batch.digital_objects[1]).to receive(:pre_assemble)
+        allow(batch.digital_objects[0]).to receive(:pre_assemble).and_return({ pre_assem_finished: false, status: 'error', message: 'oops' })
+        allow(batch.digital_objects[1]).to receive(:pre_assemble).and_return({ pre_assem_finished: true, status: 'success' })
       end
 
-      let(:yaml) { YAML.load_file(batch.progress_log_file) }
+      let(:yaml) { YAML.load_stream(File.read(batch.progress_log_file)) }
 
-      it 'logs incomplete_status error' do
-        batch.process_digital_objects
-        expect(yaml[:status]).to eq batch.send(:incomplete_status)[:status]
-        expect(yaml[:message]).to eq batch.send(:incomplete_status)[:message]
+      it 'logs an error and returns false' do
```

Comment:
change this description?  'logs success or errors as appropriate' ?

Reply:
t think this is the appropriate description for this particular test ... it is confirming that when at least one object fails, you end in an error state